components/renku_data_services/project/db.py

```diff
@@ -133,21 +132,17 @@ async def get_projects(
         async with self.session_maker() as session:
             stmt = select(schemas.ProjectORM)
             stmt = stmt.where(schemas.ProjectORM.id.in_(project_ids))
-            stmt = stmt.limit(per_page).offset(offset)
+            stmt = stmt.limit(pagination.per_page).offset(pagination.offset)
             stmt = stmt.order_by(schemas.ProjectORM.creation_date.desc())
-            result = await session.execute(stmt)
-            projects_orm = result.scalars().all()
-
-            stmt_count = select(func.count()).select_from(schemas.ProjectORM)
-            result = await session.execute(stmt_count)
-            n_total_elements = cast(int, result.scalar() or 0)
-            total_pages, remainder = divmod(n_total_elements, per_page)
-            if remainder:
-                total_pages += 1
+            stmt_count = (
+                select(func.count()).select_from(schemas.ProjectORM).where(schemas.ProjectORM.id.in_(project_ids))
+            )
 
-            pagination = PaginationResponse(page, per_page, n_total_elements, total_pages)
+            results = await gather(session.execute(stmt), session.execute(stmt_count))
+            projects_orm = results[0].scalars().all()
+            n_total_elements = cast(int, results[1].scalar() or 0)
```

Comment:
```suggestion
            total_elements = cast(int, results[1].scalar() or 0)
```
or elements_count. no need for the n